File: ngDesk-UI/src/app/models/discussion-message.ts
import { DiscussionAttachment } from './discussion-attachment';
import { DiscussionSender } from './discussion-sender';

export class DiscussionMessage {

  public MESSAGE: string;
  public MODULE: string;
  public CHAT_ID?: string;
  public COMPANY_UUID: string;
  public ENTRY_ID: string;
  public DATE_CREATED: Date;
  public SENDER: DiscussionSender;
  public ATTACHMENTS: DiscussionAttachment[];

  constructor(message, moduleId, chatId, companyUuid,
    entryId, dateCreated, sender, attachments) {
    this.MESSAGE = message;
    this.MODULE = moduleId;
    this.CHAT_ID = chatId;
    this.COMPANY_UUID = companyUuid;
    this.ENTRY_ID = entryId;
    this.DATE_CREATED = dateCreated;
    this.SENDER = sender;
    this.ATTACHMENTS = attachments;
  }

}

File: ngDesk-Company-Service/src/main/java/com/ngdesk/company/module/dao/DiscussionMessage.java
package com.ngdesk.company.module.dao;

import java.sql.Timestamp;
import java.util.Date;

import org.springframework.data.mongodb.core.mapping.Field;

import com.fasterxml.jackson.annotation.JsonProperty;

public class DiscussionMessage {

	@Field("CHAT_ID")
	@JsonProperty("CHAT_ID")
	private String chatId;

	@Field("MESSAGE")
	@JsonProperty("MESSAGE")
	private String message;

	@Field("SENDER")
	@JsonProperty("SENDER")
	private Sender sender;

	@Field("MODULE")
	@JsonProperty("MODULE")
	private String moduleId;

	@Field("DATE_CREATED")
	@JsonProperty("DATE_CREATED")
	private Date dateCreated = new Date();

	public DiscussionMessage() {

	}

	public DiscussionMessage(String chatId, String message, Sender sender, String moduleId, Date dateCreated) {
		super();
		this.chatId = chatId;
		this.message = message;
		this.sender = sender;
		this.moduleId = moduleId;
		this.dateCreated = dateCreated;
	}

	public String getChatId() {
		return chatId;
	}

	public void setChatId(String chatId) {
		this.chatId = chatId;
	}

	public String getMessage() {
		return message;
	}

	public void setMessage(String message) {
		this.message = message;
	}

	public Sender getSender() {
		return sender;
	}

	public void setSender(Sender sender) {
		this.sender = sender;
	}

	public String getModuleId() {
		return moduleId;
	}

	public void setModuleId(String moduleId) {
		this.moduleId = moduleId;
	}

	public Date getDateCreated() {
		return dateCreated;
	}

	public void setDateCreated(Date dateCreated) {
		this.dateCreated = dateCreated;
	}

}
